include/RevCoProc.h

```diff
@@ -0,0 +1,154 @@
+//
+// _RevCoProc_h_
+//
+// Copyright (C) 2017-2023 Tactical Computing Laboratories, LLC
+// All Rights Reserved
+// contact@tactcomplabs.com
+//
+// See LICENSE in the top level directory for licensing details
+//
+
+#ifndef _SST_REVCPU_REVCOPROC_H_
+#define _SST_REVCPU_REVCOPROC_H_
+
+// -- C++ Headers
+#include <ctime>
+#include <vector>
+#include <list>
+#include <algorithm>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <random>
+#include <queue>
+
+// -- SST Headers
+#include <sst/core/sst_config.h>
+#include <sst/core/output.h>
+#include <sst/core/subcomponent.h>
+
+// -- RevCPU Headers
+#include "RevOpts.h"
+#include "RevFeature.h"
+#include "RevMem.h"
+#include "RevInstTable.h"
+
+namespace SST::RevCPU {
+  class RevCoProc;
+}
+
+using namespace SST::RevCPU;
+
+namespace SST {
+  namespace RevCPU {
+
+    // ----------------------------------------
+    // RevCoProc
+    // ----------------------------------------
+    class RevCoProc : public SST::SubComponent {
+    public:
+
+      SST_ELI_REGISTER_SUBCOMPONENT_API(SST::RevCPU::RevCoProc)
+      SST_ELI_DOCUMENT_PARAMS({ "verbose", "Set the verbosity of output for the attached co-processor", "0" }
+      )
+
+      /// RevCoProc: default constructor
+      RevCoProc( ComponentId_t id, Params& params);
+
+      /// RevCoProc: default destructor
+      virtual ~RevCoProc();                               /// RevCoProc: Destructor
+      virtual bool IssueInst(RevFeature *F, RevRegFile *R, RevMem *M, uint32_t Inst) = 0;          /// RevCoProc: Instruction Interface to RevCore
+      virtual bool Reset() = 0;                           /// RevCoProc: Reset - called on startup
+      virtual bool Teardown() = 0;                        /// RevCoProc: Teardown - called when associated RevProc completes 
+      virtual bool ClockTick(SST::Cycle_t cycle) = 0;     /// RevCoProc: Clock - can be called by SST or by overriding RevCPU
+
+    protected:
+      SST::Output *output;                                ///< RevCoProc: sst output object
+    }; // class RevCoProc
+
+    // ----------------------------------------
+    // RevSimpleCoProc
+    // ----------------------------------------
+    class RevSimpleCoProc : public RevCoProc{
+    public:
+      SST_ELI_REGISTER_SUBCOMPONENT_DERIVED(RevSimpleCoProc, "revcpu",
```

Comment:
`SST_ELI_REGISTER_SUBCOMPONENT_DERIVED` is deprecated in SST 13.0.0.  This should be `SST_ELI_REGISTER_SUBCOMPONENT`

Reply:
Fixed in fc635e3